app/models/search_builder.rb

```diff
@@ -3,10 +3,64 @@
 class SearchBuilder < Blacklight::SearchBuilder
   include Blacklight::Solr::SearchBuilderBehavior
 
-  self.default_processor_chain += [:show_works_only]
+  self.default_processor_chain += %i(
+    restrict_search_to_works
+    apply_gated_discovery
+    log_solr_parameters
+  )
 
-  def show_works_only(solr_parameters)
+  def restrict_search_to_works(solr_parameters)
     solr_parameters[:fq] ||= []
     solr_parameters[:fq] << 'model_ssi:Work'
   end
+
+  def apply_gated_discovery(solr_parameters)
+    return if current_user.admin?
+
+    solr_parameters[:fq] ||= []
+    solr_parameters[:fq] << gated_discovery_filters.compact.join(' OR ')
+  end
+
+  def log_solr_parameters(solr_parameters)
+    Rails.logger.debug("Solr parameters: #{solr_parameters.inspect}")
```

Comment:
May want to remove this trace?

Reply:
@rschenk I took this from the Blacklight Access Control gem code. It comes in very handy when you're trying to debug solr query issues, for example when a particular record isn't showing up in the display results. It wouldn't be needed in production, unless we enabled debug-level logging. Are you suggesting there's a better way to do this?

Reply:
Nah I just wasn't sure if it was a mistake left over from debugging 👍